File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        int left = 0; 
        int right = 0; 

        unordered_set<int> window;

        while (right < nums.size()) {

            if (right > k ) {
                window.erase(nums[left]);
                left++;
            }

            if (window.contains(nums[right])) {
                return true;
            }

            window.insert(nums[right]);
            right++;
        }

        return false;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        var set = new HashSet<Integer>(nums.length);

        for (int num : nums) {
            if (set.contains(num)) {
                return true;
            }
            set.add(num);
        }

        return false;
    }
}